lib/iris/analysis/_regrid.py

```diff
@@ -909,6 +912,7 @@ def __call__(self, src):
             in the horizontal dimensions will be combined before regridding.
 
         """
+        # TODO: consider refactoring to allow use of @iris.util.lift_empty_masks.
```

Comment:
I did actually want to decorate this method, since it's doing a lot of heavy lifting.

I've added some more explanation to the `# TODO`: 76f7b1400

It would perhaps be even better to refactor to be more like the curvilinear regridder, but that isn't what I was suggesting in this case.